File: Housefy/src/main/java/ca/quantum/quants/it/housefy/SplashScreenActivity.java
package ca.quantum.quants.it.housefy;

/*
 * @author Artem Tsurkan, 01414146
 * @author Wenyuan Yu, N01403697
 * @author Kyrylo Lvov, n01414058
 * @course Software Project - CENG-322-0NA
 */

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import androidx.appcompat.app.AppCompatActivity;

public class SplashScreenActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTheme(R.style.Theme_Housefy_NoActionBar);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);

        new Handler(Looper.getMainLooper()).postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent intent = new Intent(SplashScreenActivity.this, MainActivity.class);
                startActivity(intent);
                finish();
            }
        }, 3000);  // delay for 3 seconds (3000 milliseconds)
    }
}

File: housefy/src/main/java/ca/quantum/quants/it/housefy/SplashScreenActivity.kt
package ca.quantum.quants.it.housefy

/*
 * @author Artem Tsurkan, n01414146
 * @author Wenyuan Yu, n01403697
 * @author Kyrylo Lvov, n01414058
 * @course Software Project - CENG-322-0NA
 */

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import ca.quantum.quants.it.housefy.components.splash.LogoAndDescription

@SuppressLint("CustomSplashScreen")
class SplashScreenActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            LogoAndDescription()
        }

        Handler(Looper.getMainLooper()).postDelayed({
            val activity = Intent(this@SplashScreenActivity, MainActivity::class.java)
            startActivity(activity)
            finish()
        }, 3000)
    }
}
